addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -113,11 +113,10 @@ def _apply_timesheet_label(self, view_arch):
     # ----------------------------------------------------
 
     def _timesheet_get_portal_domain(self):
-        return ['|', '&',
+        return ['&',
                 ('task_id.project_id.privacy_visibility', '=', 'portal'),
```

Comment:
For the record, this line represents the biggest performance hit on large databases, because it is translated by the ORM into the list of IDs of *all tasks/issues* of *all projects* with "portal" visibility. Even if the user only sees their own in the end, thanks to the rest of the domain.

Forcing `auto_join=True` on intermediary relationships improves the situation by an order of magnitude, but it is risky (bypass ACLs + overrides) and generally overkill, as the problem only occurs on large databases. An ORM improvement is planned for v14 to execute such domains via subqueries - it should bring the same performance gain without the risks/security implications.